addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -0,0 +1,261 @@
+/** @odoo-module **/
+
+import { registry } from '@web/core/registry';
+import { useService } from '@web/core/utils/hooks';
+import core from 'web.core';
+import { WebsiteEditorComponent } from '../../components/editor/editor';
+import { WebsiteTranslator } from '../../components/translator/translator';
+import {OptimizeSEODialog} from '@website/components/dialog/seo';
+
+const { Component, onWillStart, useRef, useEffect, useState } = owl;
+
+class BlockIframe extends Component {
+    setup() {
+        this.websiteService = useService('website');
+        this.state = useState({
+            blockIframe: false,
+            showLoader: false,
+        });
+        this.iframeLocks = 0;
+        this.websiteService.bus.addEventListener("BLOCK", this.block.bind(this));
+        this.websiteService.bus.addEventListener("UNBLOCK", this.unblock.bind(this));
+    }
+    block(event) {
+        if (event.detail.showLoader && !this.state.showLoader) {
+            setTimeout(() => {
+                this.state.showLoader = true;
+            }, event.detail.loaderDelay);
+        }
+        if (this.iframeLocks === 0) {
+            this.state.blockIframe = true;
+        }
+        this.iframeLocks++;
+    }
+    unblock() {
+        this.iframeLocks--;
+        if (this.iframeLocks < 0) {
+            this.iframeLocks = 0;
+        }
+        if (this.iframeLocks === 0) {
+            this.state.blockIframe = false;
+            this.state.showLoader = false;
+        }
+    }
+}
+BlockIframe.template = 'website.BlockIframe';
+
+export class WebsitePreview extends Component {
+    setup() {
+        this.websiteService = useService('website');
+        this.dialogService = useService('dialog');
+        this.title = useService('title');
+        this.user = useService('user');
+
+        this.iframeFallbackUrl = '/website/iframefallback';
+
+        this.iframe = useRef('iframe');
+        this.iframefallback = useRef('iframefallback');
+        this.container = useRef('container');
+        this.websiteContext = useState(this.websiteService.context);
+
+        onWillStart(async () => {
+            await this.websiteService.fetchWebsites();
+            if (this.websiteDomain && this.websiteDomain !== window.location.origin) {
+                window.location.href = `${this.websiteDomain}/web#action=website.website_preview&path=${this.path}&website_id=${this.websiteId}`;
```

Comment:
here too?

Reply:
Indeed. URL with multiple params were just not able to work at all without this (e.g. `/aaa?bbb=ccc&ddd=eee`). @younn-o We'll need tests for all of this later on. But we'll review everything anyway once we implement the URL to the backend action differently (in guru meeting we spoke a dedicated controller and not displaying the iframe URL in your browser. We are making specs with @rdeodoo about this.